engine/src/main/java/org/pentaho/di/repository/filerep/KettleFileRepository.java

```
@@ -0,0 +1,1571 @@
+              list.add( transName );
+            }
+          }
+        }
+      }
+
+      return list.toArray( new String[list.size()] );
+    } catch ( Exception e ) {
+      throw new KettleException(
+        "Unable to get list of transformations names in folder with id : " + id_directory, e );
+    }
+  }
+
+  public ObjectId[] getTransformationPartitionSchemaIDs( ObjectId id_transformation ) throws KettleException {
+    return new ObjectId[] {};
+  }
+
+  public String[] getTransformationsUsingCluster( ObjectId id_cluster ) throws KettleException {
+    return new String[] {};
+  }
+
+  @Override
+  public String[] getTransformationsUsingDatabase( ObjectId id_database ) throws KettleException {
+    return new String[] {};
+  }
+
+  public String[] getTransformationsUsingPartitionSchema( ObjectId id_partition_schema ) throws KettleException {
+    return new String[] {};
+  }
+
+  public String[] getTransformationsUsingSlave( ObjectId id_slave ) throws KettleException {
+    return new String[] {};
+  }
+
+  @Override
+  public String getVersion() {
+    return FILE_REPOSITORY_VERSION;
+  }
+
+  public ObjectId insertClusterSlave( ClusterSchema clusterSchema, SlaveServer slaveServer ) throws KettleException {
+    return null;
+  }
+
+  @Override
+  public void insertJobEntryDatabase( ObjectId id_job, ObjectId id_jobentry, ObjectId id_database ) throws KettleException {
+  }
+
+  public void insertJobNote( ObjectId id_job, ObjectId id_note ) throws KettleException {
+  }
+
+  @Override
+  public ObjectId insertLogEntry( String description ) throws KettleException {
+    String logfile = calcDirectoryName( null ) + LOG_FILE;
+    try {
+      OutputStream outputStream = KettleVFS.getOutputStream( logfile, true );
+      outputStream.write( description.getBytes() );
+      outputStream.write( Const.CR.getBytes() );
+      outputStream.close();
+
+      return new StringObjectId( logfile );
+    } catch ( IOException e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)